src/colorizer/loaders/UrlArrayLoader.ts

```diff
@@ -44,18 +39,12 @@ export class UrlArraySource<T extends FeatureDataType> implements ArraySource<T>
 }
 
 export default class UrlArrayLoader implements IArrayLoader {
-  private workerPool: workerpool.Pool;
+  private workerPool: SharedWorkerPool;
+  private cleanupWorkerPoolOnDispose: boolean;
 
-  constructor() {
-    // TODO: Maintain a single worker pool for all loaders/all asynchronous operations in the app?
-    this.workerPool = workerpool.pool(WorkerUrl, {
-      workerOpts: {
-        // Set worker type to undefined (classic) in production to fix a Vite issue where the application
-        // crashes when loading a module worker.
-        // Copied from https://github.com/josdejong/workerpool/tree/master/examples/vite.
-        type: import.meta.env.PROD ? undefined : "module",
-      },
-    });
+  constructor(workerPool?: SharedWorkerPool) {
+    this.cleanupWorkerPoolOnDispose = workerPool === undefined;
+    this.workerPool = workerPool ?? new SharedWorkerPool();
```

Comment:
idk if it's the right pattern for the `SharedWorkerPool` ecosystem you're looking to build, but an alternative approach worth keeping in mind for this might be reference counting the worker pool, like:

```typescript
class SharedWorkerPool {
  private workerPool: workerpool.Pool;
  private references: number;

  addReference(): void {
    this.references++;
  }

  removeReference(): void {
    this.references--;
    if (this.references <= 0) {
      this.workerPool.terminate();
    }
  }
}

class SomeSharedWorkerPoolUser {
  private workerPool: SharedWorkerPool;

  constructor(pool: SharedWorkerPool) {
    this.workerPool = pool;
    pool.addReference();
  }

  dispose(): void {
    pool.removeReference();
  }
}
```

The interface in that second class could even be made an `extends`-able base class for `SharedWorkerPool` sharers?

All optional again though. It seems just as possible from what little I know that the existing pattern, where the first class to use the workerpool effectively owns it and takes care of getting rid of it, is what makes the most sense here.

Reply:
I'll definitely keep this in mind! I think the goal for me is to have one SharedWorkerPool instance that's owned by the top-level app and is active throughout its lifetime, so it should never need to be garbage collected. I mostly have the disposal handling for testing purposes, though I don't think I have unit tests for `UrlArrayLoader` so that may be moot anyways.